include/podio/UserDataCollection.h

```diff
@@ -0,0 +1,135 @@
+#ifndef USERDATACOLLECTION_H
+#define USERDATACOLLECTION_H
+
+#include "podio/CollectionBase.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+#include <typeindex>
+
+
+namespace podio {
+
+  template <typename BasicType>
+  std::string userDataTypeName() { throw std::runtime_error( std::string(" unsupported type for UserDataCollection: ")
+							     + typeid(BasicType).name() ) ; }
+
+  template <>  std::string userDataTypeName<int>()   {return "int" ; }
+  template <>  std::string userDataTypeName<float>() {return "float" ; }
+  template <>  std::string userDataTypeName<long>()  {return "long" ; }
+  template <>  std::string userDataTypeName<double>(){return "double" ; }
```

Comment:
These could be made into a
```cpp
template<typename BasicType, 
         typename = EnableIfSupportedUserType<BasicType>> 
constexpr const char* userDataTypeName();

template<> constexpr const char* userDataTypeName<int>() { return "int"; }
// ... etc.
```
without any changes necessary to the rest of the code. Removing the implementation for the default case would also lead to a linking error if anyone tries to call this with an unsupported type. To ease the adding of additional names one could introduce a small macro, but still with that we would have to spell out the list of supported times a second time (after the initial `std::tuple`).
```cpp
#define ADD_USER_TYPE(type) template<> constexpr const char* userDataTypeName<type>( return #type; }
```